glue_lab/glue_session.py

```diff
@@ -0,0 +1,222 @@
+                            for key, value in state.items():
+                                try:
+                                    setattr(table.state, key, value)
+                                except Exception:
+                                    pass
+                    saved_viewer["rendered"] = True
+
+    def _read_view_state(
+        self, tab_name: str, viewer_id: str
+    ) -> Tuple[Optional[str], Dict]:
+        """Generate the view state from viewer id and tab name
+
+        Args:
+            tab_name (str): Name of the tab
+            viewer_id (str): Id of the viewer
+
+        Returns:
+            Tuple[Optional[str], Dict]: Viewer type and the state of the
+            viewer
+        """
+        state = {}
+        tab_data = self._document.get_tab_data(tab_name)
+        contents = self._document.contents
+        if tab_data is None:
+            return state
+
+        viewer_data = tab_data.get(viewer_id, {})
+        view_type: str = viewer_data.get("_type")
+        state_values = viewer_data.get("state", {}).get("values", {})
+        # Extract plot state
+        for prop, value in state_values.items():
+            if isinstance(value, str) and value.startswith("st__"):
+                state[prop] = value[4:]
+            else:
+                state[prop] = value
+
+        # Merging the state with what's specified in "layers"
+        # Only taking the state of the first layer
+        # TODO Support multiple layers??
+        layers = viewer_data.get("layers", [])
+        if len(layers) > 0 and layers[0].get("state") in contents:
+            extra_state = contents.get(layers[0].get("state"), {}).get("values", {})
+            for prop, value in extra_state.items():
+                if isinstance(value, str) and value.startswith("st__"):
+                    state[prop] = value[4:]
+                else:
+                    state[prop] = value
+        return view_type, state
+
+    def _init_ydoc(self) -> None:
+        """Initialize the `YGlue` document and populate its contents
+        by using the `ypywidgets.Widget`
+        """
+
+        self._sessionYDoc = Y.YDoc()
+        # Import `glue_lab.glue_ydoc.YGlue`` class through `jupyter_ydoc``
+        self._document: YGlue = ydocs.get("glu")(self._sessionYDoc)
+        self._document.observe(self._on_document_change)
+        self._ywidget = Widget(
+            comm_metadata=dict(
+                ymodel_name="GlueSession", create_ydoc=False, path=self._path
+            ),
+            ydoc=self._sessionYDoc,
+        )
+
+    def _load_data(self) -> None:
+        """Load data defined in the glue session"""
+        data_paths = {}
+        contents = self._document.contents
+        session_path = Path(self._path).parent
+        if "LoadLog" in contents:
+            path = Path(contents["LoadLog"]["path"])
+            data_paths[path.stem] = str(session_path / path)
+        idx = 0
+        while f"LoadLog_{idx}" in contents:
```

Comment:
Maybe a `while True` would be better, that would allow to not repeat `f"LoadLog_{idx}"`?